sys/riot_hdr/riot_hdr.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2017 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_riot_hdr
+ * @{
+ *
+ * @file
+ * @brief       RIOT header helpers and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#include <string.h>
+#include <stddef.h>
+
+#ifdef RIOT_VERSION
+#include "log.h"
+#else
+#include <stdio.h>
+#define LOG_INFO(...) printf(__VA_ARGS__)
+#endif
+
+#include "riot_hdr.h"
+#include "checksum/fletcher32.h"
+#include "byteorder.h"
+
+#if __BYTE_ORDER__ != __ORDER_LITTLE_ENDIAN__
+#   error "This code is implementented in a way that it will only work for little-endian systems!"
```

Comment:
After talking with @cladmi and @emmanuelsearch offline, I'm resolving this because basically there are well explained reasons for it being here:
- It's common to define a serialization format that the platform must support (e.g. also unicode/utf-8/etc for text files)
- most of our platforms are little-endian anyway